contrib/init/bitcoind.openrc

```diff
@@ -17,21 +17,28 @@ BITCOIND_GROUP=${BITCOIND_GROUP:-bitcoin}
 BITCOIND_BIN=${BITCOIND_BIN:-/usr/bin/bitcoind}
 BITCOIND_NICE=${BITCOIND_NICE:-${NICELEVEL:-0}}
 BITCOIND_OPTS="${BITCOIND_OPTS:-${BITCOIN_OPTS}}"
+: ${BITCOIND_STARTUPFILE:=${BITCOIND_PIDFILE%.pid}.startup}
 
 name="Bitcoin Core Daemon"
 description="Bitcoin cryptocurrency P2P network daemon"
 
-command="/usr/bin/bitcoind"
-command_args="-pid=\"${BITCOIND_PIDFILE}\" \
-		-conf=\"${BITCOIND_CONFIGFILE}\" \
-		-datadir=\"${BITCOIND_DATADIR}\" \
+command="/usr/bin/flock"
+command_args="--no-fork
+		$(/usr/bin/printf '%q' "${BITCOIND_STARTUPFILE}") \
+		/usr/bin/bitcoind
+		$(/usr/bin/printf '%s=%q ' \
+			-pid "${BITCOIND_PIDFILE}" \
+			-conf "${BITCOIND_CONFIGFILE}" \
+			-datadir "${BITCOIND_DATADIR}" \
+			-startupnotify "flock -u 3") \
```

Comment:
Hmm, one side effect of using `startupnotify` here is that the user can no longer use it himself...

Maybe we need to fix this to allow multiple `-*notify` commands?

Reply:
I definitely think the `-*notify` options should stack, but I don't think that's a big concern with respect to this PR. If a user wants something to happen when bitcoind is ready to handle RPC requests, they would just create an initscript that depends on `bitcoind` and let OpenRC handle the sequencing.